src/assets/TokenDetectionController.ts

```diff
@@ -96,25 +106,57 @@ export class TokenDetectionController extends BaseController<
       networkType: MAINNET,
       selectedAddress: '',
       tokens: [],
+      disabled: true,
+      chainId: NetworksChainId.mainnet,
     };
     this.initialize();
     this.getTokensState = getTokensState;
     this.getTokenListState = getTokenListState;
-    this.addTokens = addTokens;
+    this.addDetectedTokens = addDetectedTokens;
     onTokensStateChange(({ tokens }) => {
       this.configure({ tokens });
     });
 
-    onPreferencesStateChange(({ selectedAddress }) => {
-      const actualSelectedAddress = this.config.selectedAddress;
-      if (selectedAddress !== actualSelectedAddress) {
-        this.configure({ selectedAddress });
+    onPreferencesStateChange(({ selectedAddress, useTokenDetection }) => {
+      const prevDisabled = this.config.disabled;
+      const isSelectedAddressChanged =
+        selectedAddress !== this.config.selectedAddress;
+      const isTokenDetectionSupported = isTokenDetectionSupportedForNetwork(
+        this.config.chainId,
+      );
+      const isDetectionEnabled = useTokenDetection && isTokenDetectionSupported;
+      this.configure({ selectedAddress, disabled: !isDetectionEnabled });
```

Comment:
This would _turn on_ token detection when it was previously disabled. This is not good.

Currently on the extension we disable all polling when the UI is closed, using the `stop` method when the UI closes and `start` when it opens again. The `disabled` state tracks whether we _want_ the polling to be running, which right now is determined by the UI (but in the future might be determined by other factors). But here we're overriding that state, so the intent is lost.

Reply:
Tracking both token detection network support and toggle from preferences separate from disabled now. All of the listeners will also respect the disabled state.